test/gtest/ucp/test_ucp_am.cc

```diff
@@ -692,6 +700,69 @@ class test_ucp_am_nbx : public test_ucp_am_base {
     ucp_mem_h                       m_rx_memh;
 };
 
+class test_ucp_am_id : public test_ucp_am_nbx {
+protected:
+    void reset_counters()
+    {
+        test_ucp_am_nbx::reset_counters();
+        m_recv_counter_cb_1 = 0;
+        m_recv_counter_cb_2 = 0;
+    }
+
+    void test_am_id_handler()
+    {
+        reset_counters();
+
+        EXPECT_UCS_OK(set_am_data_handler_internal(
```

Comment:
can were just use set_am_data_handler?

Reply:
set_am_data_handler only receives uint16_t as a handler id (this is the original implementation, and it was suggested to keep it that way although now I am not sure why), and I wanted to reproduce a scenario where the user passes 2 different ids that are eventually translated to the same handler function, and for that I need to use a function that receives unsigned int.

Reply:
I think we need to change set_am_data_handler to accept unsigned am_im

Reply:
Another difference I forgot to mention here is that set_am_data_handler_internal doesn't check/expect success while set_am_data_handler does. All the other functions that call set_am_data_handler assume it succeeds except for this function (test_am_id_handler) so this is another reason for dividing it into 2 functions.